common/djangoapps/enrollment/views.py

```diff
@@ -729,6 +731,10 @@ def post(self, request):
                     enrollment_attributes=enrollment_attributes
                 )
 
+            cohort_name = request.data.get('cohort')
+            if cohort_name is not None:
+                cohort = get_cohort_by_name(course_id, cohort_name)
```

Comment:
I think this call will raise an exception if a cohort with this name doesn't exist, so you might want to wrap this in a try/except.

Reply:
Oh, haha, we're actually inside a `try` block here, it's just a bajillion lines further up in the function.